fs/copy.py

```diff
@@ -101,13 +155,14 @@ def copy_dir(src_fs, src_path, dst_fs, dst_path, walker=None):
         in ``src_fs``. Set this if you only want to consider a sub-set
         of the resources in ``src_fs``.
     :type walker: :class:`~fs.walk.Walker`
+    :returns: a list with the copied files dst_path.
```

Comment:
The idea is to keep the copy instaruction inside the copy_* method as usual.
The callback will be triggered only if the copy happens (e.g. if src is newer than dst).
Than is up to the user to define what to do in the callback.
If you look at my lastest commit in this PR branch, you can find the code already working.

Possible scenarios we can handle (code not tested):

1. print what's going on:

```
def echo(src_fs, src_path, dst_fs, dst_path):
       src_url = open_fs(src_fs).geturl(src_path)
       dst_url = open_fs(dst_fs).geturl(dst_path)
       print("file " + src_url+ " copied to " +dst_url  )

copy_fs_if_newer(source_root, target_root, walker, callback=echo)

```

2. counting:
```
num_file = 0 
def count(src_fs, src_path, dst_fs, dst_path):
     num_file += 1
     #we could also compute the total size of copied files in MB

copy_fs_if_newer(source_root, target_root, walker, callback=echo)

print("Copied " + str(num_file) + " files")
```

3. collect all copied files:
```
files_copied = []
def accumulate(src_fs, src_path, dst_fs, dst_path):
    url = open_fs(dst_fs).geturl(dst_path)
    files_copied.append(url)

copy_fs_if_newer(source_root, target_root, walker, callback=accumulate)

for f in files_copied:
    #do something with f...
```

Do this make sense?
GP